### ISSUE TITLE
Snowflake Failover Group Replication Schedule updates fail

### ISSUE THREAD

**Issue by gdelia-pm** [Snowflake-Labs/terraform-provider-snowflake#2544]
### Terraform CLI and Provider Versions

```
> .\terraform.exe version
Terraform v1.7.3
on windows_amd64
+ provider registry.terraform.io/snowflake-labs/snowflake v0.86.0

Your version of Terraform is out of date! The latest version
is 1.7.4. You can update by downloading from https://www.terraform.io/downloads.html
```

### Terraform Configuration

```terraform
resource "snowflake_failover_group" "SOURCE_FAILOVER_GROUP" {
  depends_on = [
    snowflake_database.DB,
    snowflake_role.FUNCTIONAL_ROLES,
    snowflake_warehouse.WH,
  ]

  name = "FAILOVER_GROUP1"
  object_types              = ["ACCOUNT PARAMETERS", "DATABASES", "INTEGRATIONS", "NETWORK POLICIES", "RESOURCE MONITORS", "ROLES", "USERS", "WAREHOUSES"]
  allowed_accounts          = ["TEST.CORE1", "TEST.CORE_DR1"]
  allowed_integration_types = ["SECURITY INTEGRATIONS", "STORAGE INTEGRATIONS"]
  allowed_databases         = toset([for db in snowflake_database.DB : db.name])

  replication_schedule {
    interval = 60 # minutes
  }
}

resource "snowflake_failover_group" "TARGET_FAILOVER_GROUP" {
  depends_on = [snowflake_failover_group.SOURCE_FAILOVER_GROUP]
  provider   = snowflake.dr
  name       = snowflake_failover_group.SOURCE_FAILOVER_GROUP.name
  from_replica {
    organization_name   = "TEST"
    source_account_name = "CORE1"
    name                = snowflake_failover_group.SOURCE_FAILOVER_GROUP.name
  }
}
```

### Expected Behavior

On initial creation the interval is set correct.  If I try to change the interval it fails
```
Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  ~ update in-place

Terraform will perform the following actions:

  # snowflake_failover_group.SOURCE_FAILOVER_GROUP will be updated in-place
  ~ resource "snowflake_failover_group" "SOURCE_FAILOVER_GROUP" {
        id                        = "FAILOVER_GROUP1"
        name                      = "FAILOVER_GROUP1"
      ~ object_types              = [
          - "PARAMETERS",
          + "ACCOUNT PARAMETERS",
            # (7 unchanged elements hidden)
        ]
        # (5 unchanged attributes hidden)

      ~ replication_schedule {
          ~ interval = 15 -> 60
        }
    }

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

snowflake_failover_group.SOURCE_FAILOVER_GROUP: Modifying... [id=FAILOVER_GROUP1]
╷
│ Error: 001003 (42000): SQL compilation error:
│ syntax error line 1 at position 165 unexpected 'REPLICATION_SCHEDULE'.
│
│   with snowflake_failover_group.SOURCE_FAILOVER_GROUP,
│   on failover_replication.tf line 9, in resource "snowflake_failover_group" "SOURCE_FAILOVER_GROUP":
│    9: resource "snowflake_failover_group" "SOURCE_FAILOVER_GROUP" {
│
```

### Actual Behavior

It appears the query being sent back to snowflake is incorrect for alters. 
What I see in query monitor is 
``` ALTER FAILOVER GROUP "FAILOVER_GROUP1" SET OBJECT_TYPES = INTEGRATIONS, USERS, WAREHOUSES, ROLES, NETWORK POLICIES, ACCOUNT PARAMETERS, RESOURCE MONITORS, DATABASES REPLICATION_SCHEDULE = '60 MINUTE' ALLOWED_INTEGRATION_TYPES = SECURITY INTEGRATIONS, STORAGE INTEGRATIONS ```
What I think should be sent is 
``` ALTER FAILOVER GROUP "FAILOVER_GROUP1" SET REPLICATION_SCHEDULE = '60 MINUTE'  ```

### Steps to Reproduce

1. `terraform apply`
2. update interval to a new number between 0-1440
3. terraform apply`

### How much impact is this issue causing?

Low

### Additional Information

No rush on this, ran the query manually to reflect the change.  All good here, just reporting the bug.

**Comment by sfc-gh-asawicki** [comment]
Hey @gdelia-pm. Thanks for reaching out to us.

One of the problems is the ACCOUNT PARAMETERS -> PARAMETERS problem solved in #2522 (solving #2517 that was reported also by you). It will be released as a part of 0.87.0 version.

The generated alter statement looks consistent with the Snowflake documentation (https://docs.snowflake.com/en/sql-reference/sql/alter-failover-group#syntax). I will check the correctness of the documentation next week (we align the generated SQL statements with the docs) and I will get back to you.

**Comment by gdelia-pm** [comment]
Ah, that makes sense now that I think about it.  
If it's less work for y'all I'm happy to test post release of 0.87, so you don't have to chase on this.

**Comment by sfc-gh-asawicki** [comment]
Thanks for the proposal! Unfortunately, we have to verify it either way. After the mentioned fix, for your config there should not be change for the `OBJECT_TYPES`, so it may work correctly. However, it may still be incorrect in situations where an update to `OBJECT_TYPES` is required.

**Comment by sfc-gh-asawicki** [comment]
Hey @gdelia-pm. We have released the fix as part of [v0.87.0](https://github.com/Snowflake-Labs/terraform-provider-snowflake/releases/tag/v0.87.0) release. Please follow the [migration guide](https://github.com/Snowflake-Labs/terraform-provider-snowflake/blob/v0.87.0/MIGRATION_GUIDE.md#v0860--v0870) during the update. Please confirm that the issue is resolved in the newest version. Thanks!

**Comment by gdelia-pm** [comment]
Confirmed, it's working in 0.87.  Thanks again!
